js/canvasmanager.ts

```diff
@@ -178,6 +198,7 @@ export class CanvasManager extends StrictEventEmitter<CanvasManagerEvents>() {
       }).filter(obj => obj !== null).map(obj => obj!);
     }
     this.redraw();
+    this.emit("change", ChangeType.SetContents);
```

Comment:
This is the only place where a SetContents change is emitted. I think we should just get rid of SetContents changes for the canvas, and add a comment saying how the change event isn't emitted when the content is set.